### PR TITLE
Added hpc-opa element for installing Intel Omni-Path Software

### PR DESCRIPTION
**Pull request by slawomirostrowski**

### CODE REVIEW

File: dib/hpc/elements/hpc-opa/README.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+DIB_HPC_IMAGE_TYPE
+  :Required: no
+  :Default:
+  :Description: Tells diskimage builder type of image to be build, HPC sms node image or
+    compute node image. Element installs different binaries for sms node and compute node.
+    Valid values are
+    compute: Builds image for HPC compute node
+    sms: Builds image for HPC sms node.
+
+DIB_HPC_OPA_PKG
```
Comment (sunilmahawar) [inline dib/hpc/elements/hpc-opa/README.rst:19]: can you please example here? And I think this parameter is mandatory, without this we can not install OPA. so you should check if this is provided and fail if it is no.

Comment (slawomirostrowski) [reply dib/hpc/elements/hpc-opa/README.rst:19]: Yes you are right it is mandatory. It is used in dib/hpc/elements/hpc-opa/source-repository-intel-opa and fails with this kind of error if not defined: 
/tmp/dib_build.6NbLbqDk/hooks/extra-data.d/98-source-repositories: line 49: DIB_HPC_OPA_PKG: unbound variable
Should we handle the error in different way? Or is the default diskimage-builder behavior ok?

Comment (sunilmahawar) [reply dib/hpc/elements/hpc-opa/README.rst:19]: you should mark "Required field" in README as yes (Line20). Right now line 20 indicates that it is not required, which is not right as per your comment.

File: dib/hpc/elements/hpc-opa/README.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  :Required: no
```
Comment (sunilmahawar) [inline dib/hpc/elements/hpc-opa/README.rst]: based on the previous comment it is required, please update this accordingly.

**Review by sunilmahawar — APPROVED** [review]

**Pull request merged**